contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -720,6 +722,55 @@ rewrite_relation_walker(Node *node, void *context)
 		return raw_expression_tree_walker(node, rewrite_relation_walker, context);
 }
 
+/*
+ * rewrite_select_relation_walker is similar to rewrite_relation_walker
+ * but an additional condition check has been added to rewrite
+ * the dbo schema name with sys for the list of sys% catalogs
+ */
+static bool
+rewrite_select_relation_walker(Node *node, void *context)
```

Comment:
How are we making sure that it is only for SELECT statements? We've just renamed the function.